src/components/MagicCodeInput.tsx

```diff
@@ -231,110 +231,108 @@ function MagicCodeInput(props) {
+    const onKeyPress = (event: Partial<NativeSyntheticEvent<TextInputKeyPressEventData>>) => {
+        const keyValue = event?.nativeEvent?.key;
         if (keyValue === 'Backspace' || keyValue === '<') {
-            let numbers = decomposeString(props.value, props.maxLength);
+            let numbers = decomposeString(value, maxLength);
 
             // If keyboard is disabled and no input is focused we need to remove
             // the last entered digit and focus on the correct input
-            if (props.isDisableKeyboard && focusedIndex === undefined) {
+            if (isDisableKeyboard && focusedIndex === undefined) {
                 const indexBeforeLastEditIndex = editIndex === 0 ? editIndex : editIndex - 1;
 
                 const indexToFocus = numbers[editIndex] === CONST.MAGIC_CODE_EMPTY_CHAR ? indexBeforeLastEditIndex : editIndex;
-                inputRefs.current[indexToFocus].focus();
-                props.onChangeText(props.value.substring(0, indexToFocus));
+                const formElement = inputRefs.current as HTMLFormElement | null;
+                (formElement?.[indexToFocus] as HTMLInputElement).focus();
+                onChangeTextProp(value.substring(0, indexToFocus));
 
                 return;
             }
 
             // If the currently focused index already has a value, it will delete
             // that value but maintain the focus on the same input.
-            if (numbers[focusedIndex] !== CONST.MAGIC_CODE_EMPTY_CHAR) {
+            if (focusedIndex !== undefined && numbers?.[focusedIndex] !== CONST.MAGIC_CODE_EMPTY_CHAR) {
                 setInput(TEXT_INPUT_EMPTY_STATE);
-                numbers = [...numbers.slice(0, focusedIndex), CONST.MAGIC_CODE_EMPTY_CHAR, ...numbers.slice(focusedIndex + 1, props.maxLength)];
+                numbers = [...numbers.slice(0, focusedIndex), CONST.MAGIC_CODE_EMPTY_CHAR, ...numbers.slice(focusedIndex + 1, maxLength)];
                 setEditIndex(focusedIndex);
-                props.onChangeText(composeToString(numbers));
+                onChangeTextProp(composeToString(numbers));
                 return;
             }
 
-            const hasInputs = _.filter(numbers, (n) => ValidationUtils.isNumeric(n)).length !== 0;
+            const hasInputs = numbers.filter((n) => ValidationUtils.isNumeric(n)).length !== 0;
 
             // Fill the array with empty characters if there are no inputs.
             if (focusedIndex === 0 && !hasInputs) {
-                numbers = Array(props.maxLength).fill(CONST.MAGIC_CODE_EMPTY_CHAR);
+                numbers = Array(maxLength).fill(CONST.MAGIC_CODE_EMPTY_CHAR);
 
                 // Deletes the value of the previous input and focuses on it.
-            } else if (focusedIndex !== 0) {
-                numbers = [...numbers.slice(0, Math.max(0, focusedIndex - 1)), CONST.MAGIC_CODE_EMPTY_CHAR, ...numbers.slice(focusedIndex, props.maxLength)];
+            } else if (focusedIndex && focusedIndex !== 0) {
```

Comment:
any specific reason we wanted to check if it is defined?

Reply:
If I remove it, there will be an error on the next line because of this `focusedIndex - 1` 😄

Reply:
Thanks.